src/frustum_models/three_dimensional_lidar_frustum.cpp

```diff
@@ -0,0 +1,138 @@
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Steve Macenski (steven.macenski@simberobotics.com)
+ *********************************************************************/
+
+#include <spatio_temporal_voxel_layer/frustum_models/three_dimensional_lidar_frustum.hpp>
+
+namespace geometry
+{
+
+/*****************************************************************************/
+ThreeDimensionalLidarFrustum::ThreeDimensionalLidarFrustum(const double &vFOV, const double &hFOV,
+                                                           const double &min_dist, const double &max_dist) : _vFOV(vFOV), _hFOV(hFOV), _min_d(min_dist), _max_d(max_dist)
+/*****************************************************************************/
+{
+  _valid_frustum = true;
+  ros::NodeHandle nh;
+
+  _vFOVhalf = _vFOV / 2.0;
+  _hFOVhalf = _hFOV / 2.0;
+}
+
+/*****************************************************************************/
+ThreeDimensionalLidarFrustum::~ThreeDimensionalLidarFrustum(void)
+/*****************************************************************************/
+{
+}
+
+/*****************************************************************************/
+void ThreeDimensionalLidarFrustum::TransformModel(void)
+/*****************************************************************************/
+{
+}
+
+/*****************************************************************************/
+bool ThreeDimensionalLidarFrustum::IsInside(const openvdb::Vec3d &pt)
+/*****************************************************************************/
+{
+
+  Eigen::Vector3d point_in_global_frame(pt[0], pt[1], pt[2]);
+  Eigen::Vector3d transformed_point = _orientation.conjugate() * (point_in_global_frame - _position);
+
+  double radial_distance = sqrt((transformed_point[0] * transformed_point[0]) + (transformed_point[1] * transformed_point[1]));
+
+  // Check if inside frustum valid range
+  if (radial_distance > _max_d || radial_distance < _min_d)
+  {
+    return false;
+  }
+
+  #define CONEPADDING 0.0   //This value shifts the frustum "cone" outwards without shifting its angle.
+
+  // // Check if inside frustum valid vFOV
+  if (atan((fabs(transformed_point[2]) + CONEPADDING) / radial_distance) > _vFOVhalf)
+  {
+    return false;
+  }
+
+  // Check if inside frustum valid hFOV
+  if (transformed_point[0] > 0) {
+    if (fabs(atan(transformed_point[1] / transformed_point[0])) > _hFOVhalf)
+    {
+      return false;
+    }
+  }
+  else{
+    if ( ( fabs(atan(transformed_point[0] / transformed_point[1])) + 1.570796 )  > _hFOVhalf)
```

Comment:
Can you remove the spaces between the `( ( ` or change the `if()` to include those same spaces?